sw/device/silicon_creator/rom/e2e/BUILD

```
@@ -2661,3 +2661,39 @@ test_suite(
     tags = ["manual"],
     tests = ["sigverify_usage_constraint_{}".format(t["name"]) for t in test_cases],
 )
+
+opentitan_flash_binary(
+    name = "rom_ext_upgrade_test",
+    srcs = ["rom_ext_upgrade_test.c"],
+    devices = [
+        "fpga_cw310",
+    ],
+    deps = [
+        "//hw/ip/otp_ctrl/data:otp_ctrl_regs",
+        "//sw/device/lib/testing/test_framework:ottf_main",
+        "//sw/device/silicon_creator/lib:boot_data",
+        "//sw/device/silicon_creator/lib/drivers:lifecycle",
+        "//sw/device/silicon_creator/lib/drivers:otp",
+        "//sw/device/silicon_creator/lib/drivers:rstmgr",
+        "//sw/device/lib/testing/test_framework:ottf_ld_silicon_creator_slot_a",
+    ],
+)
+
+opentitan_functest(
+    name = "rom_ext_upgrade",
+    cw310 = cw310_params(
+        # Fail if we see version 3, because the test has updated twice, FAIL, version 0 twice, or PASS
+        exit_failure = "min_security_version_rom_ext:3|FAIL|min_security_version_rom_ext:0(?s:.*)min_security_version_rom_ext:0|PASS",
```

Can you try simplifying this using a negated character class like:
```suggestion
        exit_failure = "min_security_version_rom_ext:[^01]|FAIL|PASS",
```